app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorPublishedView.java

```diff
@@ -0,0 +1,516 @@
+  private MenuItem editItem;
+  private MenuItem restartItem;
+  private MenuItem inviteItem;
+  private boolean isRequested;
+
+  private Map<String, User> commitUsers = new HashMap<>();
+
+  public FavorPublishedView() {
+    // create favor detail from a favor
+  }
+
+  @Override
+  public void onCreate(Bundle bundle) {
+    super.onCreate(bundle);
+    setHasOptionsMenu(true);
+  }
+
+  @Override
+  public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
+
+    // Inflate the menu; this adds items to the action bar if it is present.
+    inflater.inflate(R.menu.menu_favor_detail, menu);
+
+    cancelItem = menu.findItem(R.id.cancel_button);
+    editItem = menu.findItem(R.id.edit_button);
+    restartItem = menu.findItem(R.id.restart_button);
+    inviteItem = menu.findItem(R.id.share_button);
+    // if (favorStatus != null) {
+    //  updateAcceptBtnDisplay();
+    // }
+    super.onCreateOptionsMenu(menu, inflater);
+  }
+
+  // handle button activities
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    int id = item.getItemId();
+    if (id == R.id.cancel_button) {
+      if(isRequested)
+        cancelFavor();
+      else
+        cancelCommit();
+    } else if (id == R.id.edit_button) {
+      goEditFavor();
+    } else if (id == R.id.share_button) {
+      onShareClicked();
+    } else if (id == R.id.restart_button) {
+      goRestartFavor();
+    }
+    return super.onOptionsItemSelected(item);
+  }
+
+  @Override
+  public View onCreateView(
+      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    // inflate view
+    View rootView = inflater.inflate(R.layout.fragment_favor_published_view, container, false);
+    setupButtons(rootView);
+
+    toolbar = requireActivity().findViewById(R.id.toolbar_main_activity);
+    favorViewModel =
+        (IFavorViewModel)
+            new ViewModelProvider(requireActivity())
+                .get(DependencyFactory.getCurrentViewModelClass());
+    String favorId = "";
+    if (currentFavor != null) favorId = currentFavor.getId();
+    if (getArguments() != null) favorId = getArguments().getString(CommonTools.FAVOR_ARGS);
+    setupFavorListener(rootView, favorId);
+    return rootView;
+  }
+
+  public IFavorViewModel getViewModel() {
+    return favorViewModel;
+  }
+
+  private void setupFavorListener(View rootView, String favorId) {
+    getViewModel()
+        .setObservedFavor(favorId)
+        .observe(
+            getViewLifecycleOwner(),
+            favor -> {
+              try {
+                if (favor != null && favor.getId().equals(favorId)) {
+                  currentFavor = favor;
+                }
+              } catch (Exception e) {
+                Log.d(TAG, e.getMessage());
+                CommonTools.showSnackbar(rootView, getString(R.string.error_database_sync));
+                enableButtons(false);
+                throw e;
+              }
+              if(currentFavor != null)
```

Comment:
This reason is for the ease of debug. Because I want to call `displayFromFavor(rootView, currentFavor);` outside try-catch of db error, so I check same condition again. `displayFromFavor(rootView, currentFavor);` sometimes throw exception when some field of favor is null, which is irrelevant with db error handler.